### DISCUSSION TITLE
Getting app cast parse error "line 9: xmlParseEntityRef: no name"

### DISCUSSION THREAD

**Discussion by rdutoit** [sparkle-project/Sparkle#2474] [Q&A]
### Description of the problem
Hi, I've used sparkle for years but had stopped for a while and now using in my new app. did everything exactly, used generate_keys, generate_appcast tools in archive on dropbox. Always worked easily before (with older sparkle.)

### Do you use Sandboxing in your app?
NO

### Version of `Sparkle.framework` in the latest version of your app
2.5.1

### Version of `Sparkle.framework` in the old version of app that your users have (or N/A)
N/A

### Sparkle's output from Console.app

Error: An error occurred while parsing the update feed. (null) (URL (null))
Error: Line 9: xmlParseEntityRef: no name
 (null) (URL (null))

### Steps to reproduce the behavior

feed url:
https://www.dropbox.com/scl/fi/v0pdhlya9bv66r6yljxdz/alpharob.xml

Appcast:
<?xml version="1.0" standalone="yes"?>
<rss xmlns:sparkle="http://www.andymatuschak.org/xml-namespaces/sparkle" version="2.0">
    <channel>
        <title>backupList+</title>
        <item>
            <title>9.1.0</title>
            <pubDate>Tue, 28 Nov 2023 21:22:31 -0500</pubDate>
            <sparkle:version>9100</sparkle:version>
            <sparkle:shortVersionString>9.1.0</sparkle:shortVersionString>
            <sparkle:minimumSystemVersion>10.13</sparkle:minimumSystemVersion>
            <enclosure url="https://www.dropbox.com/scl/fi/rfttm6p8s85qxvgab0567/backupList_910.zip" length="5722513" type="application/octet-stream" sparkle:edSignature="HR5CX25SErd212rlRSi62OP64kgyVquPEcpsSiNv1O1/BcJp+G2xWGZ6kF+luw02VYih/pygVqw7w4y2H4TnBg=="/>
        </item>
    </channel>
</rss>

**Comment by zorgiepoo** [comment]
Very likely an issue with the appcast XML that is being downloaded. Dropbox won't allow me to view your appcast without signing up. Hopefully you aren't actually trying to serve the file on dropbox as that will likely be problematic -- dropbox is not supposed to be used for direct-link file hosting like that (to my knowledge).

**Reply by rdutoit** [reply]
I had my archive for years on Dropbox. I know others do the same but perhaps that won’t work any more…Odd you couldn’t view the xml. It’s on public viewing… but the xml is generated and very straightforward.

**Reply by rdutoit** [reply]
Sorry the full link to read appcast is

https://www.dropbox.com/scl/fi/v0pdhlya9bv66r6yljxdz/alpharob.xml?rlkey=a99bkey9j20notahp93wy8tai&dl=0

**Reply by zorgiepoo** [reply]
I can see the appcast now but it's still not a direct link to the XML file (it shows up in Dropbox's custom viewer). I don't notice anything visibly wrong with the contents from a glance.

**Reply by rdutoit** [reply]
Yes it does seem to bring up the viewer, before it would just open in browser window. I added a bunch of logging to SUAppcsst.m and now see it isn’t receiving the appcast data to parse even. I can easily get it via curl so not sure why sparkle can’t. Some Dropbox wall perhaps but it always worked seamlessly before.

**Reply by rdutoit** [reply]
Ok so I uploaded app cast to backBlaze B2 and same error. It isn't accessing the appcast data even though it can be done easily with curl. Odd

**Reply by zorgiepoo** [reply]
I don't recommend using Dropbox. If you want a free alternative maybe GitHub Pages works. In any case getting your hosting provider to work properly is a bit out of scope with support I can provide.

**Reply by rdutoit** [reply]
Thanks. Yes you were right. After adjusting the appcast correctly it now works on back blaze b2. Not on Dropbox!
